pkg/tsdb/cloudmonitoring/resource_handler.go

```diff
@@ -235,10 +235,57 @@ func processData(res *http.Response, responseFn processResponse) ([]byte, int, e
 	if err != nil {
 		return nil, http.StatusInternalServerError, fmt.Errorf("unable to encode response %v", err)
 	}
-
 	return buf.Bytes(), 0, nil
 }
 
+func processData(req *http.Request, cli *http.Client, responseFn processResponse) ([]byte, http.Header, int, error) {
```

Comment:
didn't spend too much time looking at this but this function seems to do too many things. Maybe we can split it to different ones that can be easier to test? (if needed). From a first glance, it seems that there is code to do a loop over requests, parse the result and then prepare to send the response. Maybe that's a good subdivision?

Also, `processData` does not match what this function does anymore so it requires some naming, easier if you split it :)

Reply:
I refactored a bit, subdividing it is a bit weird, since there are a lot of variables to pass